jetty-io/src/main/java/org/eclipse/jetty/io/AdapterMemoryPool.java

```diff
@@ -0,0 +1,42 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2021 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.io;
+
+import java.nio.ByteBuffer;
+import java.util.function.Consumer;
+
+public class AdapterMemoryPool implements MemoryPool<RetainableByteBuffer>
+{
+    private final ByteBufferPool byteBufferPool;
+    private final Consumer<ByteBuffer> releaser;
```

Comment:
why is this field needed?  `byteBufferPool::release` can be passed directly the the RBB

Reply:
True, I added this field because I thought directly passing `byteBufferPool::release` to `RetainableByteBuffer` would need to allocate. Now I'm not certain about this.

Reply:
hmmm maybe it does allocated.    so leave as is unless you test that it doesn't